bpf_generic/src/bpf_loading.cpp

```diff
@@ -346,7 +345,17 @@ void bpf_subsystem::load_bpf_file(const std::string& path, uint32_t map_max_entr
 		LOG_ERROR("Cannot initialize perf maps");
 
 	processReloSections(elf, &ehdr, content.sections, content.symbols, maps);
-	load_programs_from_sections(content.sections, content.license, content.kernVersion);
+
+	auto kernelVersion{getKernelVersion()};
+	if (!kernelVersion) {
+		throw std::runtime_error{"Could not obtain current kernel version"};
+	}
+	if (!isKernelSupported(*kernelVersion)) {
+		LOG_ERROR("Kernel version {} is not supported", kernelVersionToString(*kernelVersion));
+		return;
```

Comment:
Maybe to be consistent with line 351 and also throw here  or no return  and log error and   try to load bpf progam  what will fail

Reply:
Well yeah, you'll get more logs by continuing to load. I was thinking about replacing the 1-line error log with loading errors but if we combine them instead of replacing, then ok. Also, I'm not sure if on a very specific host with kernel version < 4.15 loading NetTracer would not be successful, in which case returning early would be undesired... Anyway, I'll delete the return statement.